src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
```

Comment:
my reasoning in #29 was that clients needing both kind of streams will be more convenient to use if there is one common interface. otherwise, i would need to require two arguments, which usually will be the same object twice.
multipart stream is a special kind of stream. i can't imagine a library that only implements multipart streams but not standard streams.

if that assumption is wrong then separate interfaces indeed are better.